components/usage/pkg/contentservice/client.go

```diff
@@ -20,7 +20,7 @@ import (
 )
 
 type Interface interface {
-	UploadUsageReport(ctx context.Context, filename string, report map[db.AttributionID][]db.WorkspaceInstanceForUsage) error
+	UploadUsageReport(ctx context.Context, filename string, report []db.WorkspaceInstanceUsage) error
```

Comment:
By taking the `[]db.WorkspaceInstanceUsage` we ensure that what goes to the bill is the same as our records, but also we no longer need the pricing config. The credit price has already been computed so there's no need to compute it again.

Reply:
The previous report format had the nice property that usage was keyed by attribution id, whereas now it's a flat list of instances. Does that matter?

Reply:
I think there was a higher chance of error, especially in tests The previous form was:
```
{
   "team:<teamID>": [
       { InstanceID: "asdas", AttributionID: "asdad" }
   ]
}
```
We've moved the grouping into the billing controller which I think works better, and I don't see much harm in doing the grouping there.